### ISSUE TITLE
MenuItem: split content and info prop into accessible name and accessible description

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58720]
### Description

The MenuItem component content is actually determined by its children _and_ the value of the `info` prop.

As such, both the content and the `info` end up being the actual label of this UI control, i.e. the _accessible name_.

This is less than ideal because the `info` is typically used to provide a visual _descriotion_ of the control label. However, the description is part of the _name_ of the control, thus labeling the control with extraneous information.
Generally:
- A control's accessible name should be short and sufficiently meaningful.
- An additional description may be provided _separately_ from the accessible name. It should be rendered into a separate  element with an ID that is referenced via `aria-describedby`.

Putting together the name and description provides users with a label that is semantically too long and confusing. Assistive technologies use the label (name) which in many cases is just too long and may even be a barrier for some users.

Examples:

The URL input UI for images shows some preset suggestions in a menu with menuitems. The 'Expand on click` last item was added in https://github.com/WordPress/gutenberg/pull/57608

Visually, the label 'Expand on click' and the description 'Scale the image with a lightbox effect.' are separated and style differently:

![Screenshot 2024-02-06 at 09 10 53](https://github.com/WordPress/gutenberg/assets/1682452/6c383b5e-d920-4cfa-9821-e17bc2116700)

However, that's only visuals. Both strings are the actual content of the MenuItem button. As such, the accessible name of the menu item is:

`Expand on click Scale the image with a lightbox effect.`

![Screenshot 2024-02-06 at 10 35 31](https://github.com/WordPress/gutenberg/assets/1682452/5f1d673e-c557-4669-b728-f7970e28cb44)

Aside: I don't think a descriptive text in the menu item should end with a period.

That string is less than ideal:
- Too noisy for screen reader users. 
- Too long to be easily activated with a voice command for speech recognition software users.
- Semantically not ideal as it adds a _description_ into the _name_.

There are many other cases where a menu item has descriptions that are actually rendered as  part of the _name_. For example: block alignment controls:

![Screenshot 2024-02-06 at 11 11 52](https://github.com/WordPress/gutenberg/assets/1682452/068f74a1-8c09-4ac1-b6cc-ffe628bc8341)

Options menu items, etc.:

![Screenshot 2024-02-06 at 11 13 03](https://github.com/WordPress/gutenberg/assets/1682452/fcae708e-c6de-4151-9779-9a850a6151ee)

Also in these cases, such long accessible names are less than ideal, e.g.:

`Top toolbar Access all block and document tools in a single place`

Looking at the code, it appears the only purpose of the `info` text is to provide additional text that _looks_ like a description. There's no semantics at all.

https://github.com/WordPress/gutenberg/blob/1fb5110da6444424202bcfe4d8c611aaa9594ba7/packages/components/src/menu-item/index.tsx#L42-L45

Originally introduced in https://github.com/WordPress/gutenberg/pull/10802

### Step-by-step reproduction instructions

- Inspect in your browser dev tools one of these menu items with descriptions.
- Observe the visual description is actually part of the item label (its content) thus determining the accessible name

### Screenshots, screen recording, code snippet

More screenshots of how these long accessible names are announced by a screen reader:

![Screenshot 2024-02-06 at 11 22 08](https://github.com/WordPress/gutenberg/assets/1682452/56685559-cb41-4517-b090-71b4ba1e5a98)

![Screenshot 2024-02-06 at 11 23 10](https://github.com/WordPress/gutenberg/assets/1682452/b6cb18a2-a10e-4c56-8af6-3aa4ff6f52d1)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Disabling CSS helps understand also visually these MenuItems (under the hood they are button elements) have too long labels. I'm not sure no one would ever craft buttons with such long text:

![Screenshot 2024-02-06 at 11 35 11](https://github.com/WordPress/gutenberg/assets/1682452/933c3507-277f-492a-a160-9857cdf64e4a)

![Screenshot 2024-02-06 at 11 40 07](https://github.com/WordPress/gutenberg/assets/1682452/dbf3b670-0fba-4f2f-b967-56f1c58d83de)
